crates/typst/src/layout/grid/layout.rs

```diff
@@ -270,6 +300,23 @@ impl CellGrid {
             };
             let x = resolved_index % c;
             let y = resolved_index / c;
+            let colspan = cell.colspan(styles).get();
+
+            if colspan > c - x {
+                bail!(
+                    cell_span,
+                    "cell's colspan would cause it to exceed the available column(s)";
+                    hint: "try placing the cell in another position or reducing its colspan"
+                )
+            }
+
+            let Some(largest_index) = resolved_index.checked_add(colspan - 1) else {
+                bail!(
+                    cell_span,
+                    "cell would span an exceedingly large position";
+                    hint: "try reducing the cell's colspan"
+                )
+            };
```

Comment:
Your concern is justified by the fact that this check is unlikely to ever be triggered for colspans, since adding a large amount of columns in the first place would fill memory usage very quickly. However, this check will be much more relevant when rowspans are implemented, since it'd be very easy to span enough rows to cause overflow. So I think it will end up making more sense down the road here.

Reply:
okay